File: Ticket Counter - GFG/ticket-counter.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;


// } Driver Code Ends

// class Solution {
//   public:
//     int distributeTicket(int N, int K) {
//         // code here
//         deque<int> ad;
//     for (int i = 1; i <= N; i++) {
//         ad.push_back(i);
//     }
//     int flag = 0;
//     int curr = 0;
//     while (!ad.empty()) {
//         int time = 0;
//         if (flag == 0) {
//             while (!ad.empty() && time < K) {
//                 curr = ad.front();
//                 ad.pop_front();
//                 time++;
//             }
//         } else {
//             while (!ad.empty() && time < K) {
//                 curr = ad.back();
//                 ad.pop_back();
//                 time++;
//             }
//         }
//         flag =!flag;
//     }
//     return curr;
//     }
// };


class Solution {
  public:
    int distributeTicket(int N, int K) {
        // code here
    int st=1; int end=N;
        while(st<end)
        {
            int time=K;
            while(st<end && time-- >0) st++;
            time=K;
            while(st<end && time-- >0) end--;
            
        }
        return st;
        
    }
};

//{ Driver Code Starts.

int main(){
    int t;
    scanf("%d ",&t);
    while(t--){
        
        int N;
        scanf("%d",&N);
        
        
        int K;
        scanf("%d",&K);
        
        Solution obj;
        int res = obj.distributeTicket(N, K);
        
        cout<<res<<endl;
        
    }
}

// } Driver Code Ends

File: Ticket Counter - GFG/ticket-counter.java
//{ Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*;

class GFG{
    public static void main(String args[]) throws IOException { 
        BufferedReader in=new BufferedReader(new InputStreamReader(System.in));
        PrintWriter out=new PrintWriter(System.out);
        int t = Integer.parseInt(in.readLine());
        while (t-- > 0) {
            String s[]=in.readLine().trim().split(" ");
            int n=Integer.parseInt(s[0]);
            int k=Integer.parseInt(s[1]);
            Solution ob=new Solution();
            out.println(ob.distributeTicket(n,k));
        }
        out.close();
    }
}
// } Driver Code Ends


//User function Template for Java
class Solution {
    public static int distributeTicket(int N,int K)
    {
        ArrayDeque<Integer>ad = new ArrayDeque<>();
        for(int i=1; i<=N; i++)
        {
            ad.add(i);
        }
        int flag=0;
         int curr=0;
        while(!ad.isEmpty())
        {
           
            int time=0;
            if(flag==0)
            {    while(!ad.isEmpty() && time<K)
                   {
                       curr= ad.pollFirst(); time++;
                   }
            }

            else
            {   while(!ad.isEmpty() && time<K)
                  {
                      curr= ad.pollLast(); time++;
                  }
            }

           flag  = flag==0? 1:0;
        }
        
        
         return curr;
    }
}
